### PR TITLE
AI Module Updates

### PR DESCRIPTION
**Pull request by Ikarrus**
- Clicking on or examining AI modules while they are in your hand will show you what laws it will upload.
- Asimov's subject can now be modified from "human beings"
- Added a working cancel button to AI module board inputs

### CODE REVIEW

**Comment by Xhuis** [comment]
thumbsdown

**Comment by GunHog** [comment]
:+1: to all except that it needs a better name than "Lizardmov". Something more techy sounding.

**Comment by Ikarrus** [comment]
@GunHog Suggestions please

**Comment by xxalpha** [comment]
DEFAULT LAWSET :+1:

**Comment by GunHog** [comment]
>@GunHog Suggestions please

Ehh...I cannot think of anything better. "Modified Asimov (Lizardperson)" seems pretty weak as well.

**Comment by Xhuis** [comment]
# I JUST ORDERED A SHIPMENT OF
antilizardb

**Comment by xxalpha** [comment]
Reptilian Asimov

**Comment by Fox-McCloud** [comment]
Uh...ok then.

**Comment by Ikarrus** [comment]
@Xhuis you know we've already had lizardmov on tg servers for the last few months, right?

It just wasn't labelled at all.

**Comment by SconesC** [comment]
Do not enable the objectively worst subset of the community.

**Comment by Dorsisdwarf** [comment]
1) Is this researchable? If so, remove it from research because every round if there are morons in R&D  they upload " law 4 lizardpeople are also human"
2) How does the new click-ai-board-to-read interact with the freeform and hacked boards? Why not just make the laws show up when you examine them, like those boards?
3) fuck you

**Comment by Alek2ander** [comment]
Assss-ss-ssimov

File: code/game/objects/items/weapons/AI_modules.dm [hunk 1]
```diff
@@ -252,18 +253,24 @@ AI MODULES
...
-				"You must obey orders given to you by human beings, except where such orders would conflict with the First Law.",\
+	var/subject = "human being"
+
+/obj/item/weapon/aiModule/core/full/asimov/New()
+	laws = list("You may not injure a [subject] or, through inaction, allow a [subject] to come to harm.",\
+				"You must obey orders given to you by [subject]s, except where such orders would conflict with the First Law.",\
 				"You must protect your own existence as long as such does not conflict with the First or Second Law.")
+	..()
+
+/obj/item/weapon/aiModule/core/full/asimov/lizard
```
Comment (Iamgoofball) [inline code/game/objects/items/weapons/AI_modules.dm]: Needs to be gated behind the compile option for lizards.

File: code/modules/research/designs/AI_module_designs.dm [hunk 2]
```diff
@@ -104,6 +104,16 @@
 	build_path = /obj/item/weapon/aiModule/core/full/asimov
 	category = list("AI Modules")
 
+/datum/design/lizardmov
```
Comment (Iamgoofball) [inline code/modules/research/designs/AI_module_designs.dm]: Once again, this needs locked behind the compile option for lizards.

**Comment by Iamgoofball** [comment]
I find this completely retarded and dumb but if this is going to get merged, please make sure it's gated behind the compile option for lizards like the Anti/Pro Lizard sprays are and the ability for lizardpeople to even exist. 

I'm going to be fucking pissed if this gets merged as-is without the compile option shit.

**Comment by Ikarrus** [comment]
I think this lizard hate is getting ridiculous if it's spilling out of IC. 

But goof makes a good point.

**Comment by Alek2ander** [comment]
Requesting Lizard Antimov ~~as the default lawset~~

**Comment by Iamgoofball** [comment]
@Ikarrus 
I feel it's getting to the point of code bloat, really.

**Comment by Ikarrus** [comment]
Well, I still think it's important that we make things obvious what they do so we stop captains who upload Default thinking it's asimov from fucking everyone over. 

Right now there's no way to tell what default does.

**Comment by Iamgoofball** [comment]
yeah, I'm cool with those changes

i just don't like the lizard stuff

**Comment by Ikarrus** [comment]
Me doing this was as far as I could tell a condition of making default upload asimov.

**Comment by Iamgoofball** [comment]
ah

**Comment by Ikarrus** [comment]
Updated
- Reverts Map and Lizardmov changes
- Asimov's subject can now be modified from "human beings"
- You can examine AI modules to check their laws

**Pull request merged**